lib/datadog/tracing/contrib/configuration/resolver.rb

```diff
@@ -79,6 +79,49 @@ def parse_matcher(matcher)
             matcher
           end
         end
+
+        # The {CachedResolver} is a mixin that provides caching functionality to the {Resolver} class.
+        # This is useful when {Resolver#resolve} values that are expensive to compute.
+        # This is a size-limited, FIFO cache.
+        #
+        # @example
+        #   class MyResolver < Datadog::Tracing::Contrib::Configuration::Resolver
+        #     prepend Datadog::Tracing::Contrib::Configuration::CachedResolver
+        #     # ...
+        #   end
+        module CachedResolver
```

Comment:
I would call this `CachingResolver` because the resolver is what is doing the caching, not the thing that is being cached.